pennylane/pulse/hardware_hamiltonian.py

```diff
@@ -191,89 +191,67 @@ def circuit_local(params):
 
             amplitude = lambda p, t: p * jnp.sin(jnp.pi * t)
             phase = jnp.pi / 2
-            detuning = 3 * jnp.pi / 4
-            H_d = qml.pulse.drive(amplitude, phase, detuning, wires)
 
-            dev = qml.device("default.qubit.jax", wires=wires)
+            H_d = qml.pulse.drive(amplitude, phase, wires)
 
+            dev = qml.device("default.qubit.jax", wires=wires)
             @qml.qnode(dev, interface="jax")
             def circuit(params):
                 qml.evolve(H_i + H_d)(params, t=[0, 10])
                 return qml.expval(qml.PauliZ(1))
 
         >>> params = [2.4]
         >>> circuit(params)
-        Array(0.94301294, dtype=float64)
+        Array(0.96893251, dtype=float64)
         >>> jax.grad(circuit)(params)
-        [Array(0.59484969, dtype=float64)]
-
+        [Array(-0.0355582, dtype=float64)]
     """
     if isinstance(wires, int):
         wires = [wires]
-    trivial_detuning = not callable(detuning) and qml.math.isclose(detuning, 0.0)
 
     if not callable(amplitude) and qml.math.isclose(amplitude, 0.0):
-        if trivial_detuning:
-            raise ValueError(
-                f"Expected non-zero value for at least one of either amplitude or detuning, but received amplitude={amplitude} and detuning={detuning}. All terms are zero."
-            )
-
-        coeffs = []
-        observables = []
+        raise ValueError(
+            f"Expected non-zero value or callable for amplitude, but received amplitude={amplitude}. All terms are zero."
+        )
```

Comment:
This doesn't need to be addressed in this PR, but I want to note that this check should probably happen in the `rydberg_drive` and `transmon_drive` functions. If this check is kept in `drive`, then `rydberg_drive` will unexpectedly fail if a drive term with only detuning is specified (which is possible for local drives). It would be fine if we also check for this is `rydberg_drive`, but that would lead to unnecessary redundancy.

Reply:
Good point! I suggest you remove it in `drive` in your PR